front/lib/models/assistant/conversation.ts

```diff
@@ -244,6 +244,9 @@ export class AgentMessage extends BaseModel<AgentMessage> {
   declare agentConfigurationVersion: number;
 
   declare agentMessageContents?: NonAttribute<AgentMessageContent[]>;
+  // This points to the message associated with the agent message
+  declare agentMessage?: NonAttribute<Message>;
```

Comment:
Nope this seems incorrect. Right. AgentMessage.agentMessage can't point to a Message

Reply:
I think there is a misnaming in the aliases from AgentMessage/UserMessage to Message and we need to fix them right?

Reply:
Ok then opting for the broader refactor, indeed the existing alias is off